nucypher/keystore/keypairs.py

```diff
@@ -17,26 +22,29 @@ class Keypair(object):
     A parent Keypair class for all types of Keypairs.
     """
 
+    _private_key_source = UmbralPrivateKey.gen_key
+    _public_key_method = "get_pubkey"
+
     def __init__(self,
-                 umbral_key: Union[UmbralPrivateKey, UmbralPublicKey] = None,
+                 private_key: Union[UmbralPrivateKey, UmbralPublicKey] = None,
                  generate_keys_if_needed=True):
         """
         Initalizes a Keypair object with an Umbral key object.
 
-        :param umbral_key: An UmbralPrivateKey or UmbralPublicKey
+        :param private_key: An UmbralPrivateKey or UmbralPublicKey
         :param generate_keys_if_needed: Generate keys or not?
         """
         try:
-            self.pubkey = umbral_key.get_pubkey()
-            self._privkey = umbral_key
+            self.pubkey = getattr(private_key, self._public_key_method)()
+            self._privkey = private_key
         except NotImplementedError:
-            self.pubkey = umbral_key
-            self._privkey = PUBLIC_ONLY
+            self.pubkey = private_key
+            self._privkey = constants.PUBLIC_ONLY
```

Comment:
OK, I guess this is the Pythonic way to do it, but I find it very obscure. Why can you just do the same with a couple of `isinstance`? What if, for some strange/stupid reason, we decide later in pyUmbral that `UmbralPublicKey.get_pubkey()` returns `self` (which doesn't sound that stupid after all)?